packages/xfsprogs/0001-mkfs-source-defaults-from-config-file.patch

```diff
@@ -0,0 +1,54 @@
+From 5725241ef76ee005fb75923815110fc78f2bb004 Mon Sep 17 00:00:00 2001
+From: Piyush Jena <piyushjena1996@gmail.com>
+Date: Tue, 13 May 2025 21:46:19 +0000
+Subject: [PATCH] mkfs: source defaults from config file
+
+Signed-off-by: Piyush Jena <piyushjena1996@gmail.com>
+---
+ mkfs/xfs_mkfs.c | 23 +++++++++++++++++++++++
+ 1 file changed, 23 insertions(+)
+
+diff --git a/mkfs/xfs_mkfs.c b/mkfs/xfs_mkfs.c
+index 3f4455d4..b6755a1a 100644
+--- a/mkfs/xfs_mkfs.c
++++ b/mkfs/xfs_mkfs.c
+@@ -40,6 +40,9 @@
+  */
+ #define WHACK_SIZE (128 * 1024)
+
++/* Default path for the mkfs.xfs configuration file */
++#define DEFAULT_CONFIG_PATH "/usr/share/xfsprogs/mkfs/default.conf"
++
+ /*
+  * XXX: The configured block and sector sizes are defined as global variables so
+  * that they don't need to be passed to getnum/cvtnum().
+@@ -5250,6 +5253,26 @@ main(
+ 	 * printf(_("Default configuration sourced from %s\n"), dft.source);
+ 	 */
+
++	// check if default config path exists
++	struct stat defcfg_statbuf;
++	if (stat (DEFAULT_CONFIG_PATH, &defcfg_statbuf) == 0) {
++		/*
++		* Set the sb_feat for cli_params as mentioned above. The purpose of
++		* this is to make sure we inadvertently don't override any default
++		* set by hardcoding. We use the existing config parsing function
++		* (cfgfile_parse) to read the defaults.
++		*/
++		struct cli_params defcfg_args = {
++			.cfgfile = DEFAULT_CONFIG_PATH,
```

Comment:
Does this allow a user to pass a `-c configuration_file` still or have we overwritten this behavior? Can you add some tests confirming a user could still create an XFS file system with this argument?

Reply:
Updated the description with more tests.